src/gui/scene/Renderer.js

```diff
@@ -1,15 +1,103 @@
 import * as THREE from '../../../thirdPartyCode/three/three.module.js';
 import { RoomEnvironment } from '../../../thirdPartyCode/three/RoomEnvironment.module.js';
 import { acceleratedRaycast } from '../../../thirdPartyCode/three-mesh-bvh.module.js';
+import { VRButton } from '../../../thirdPartyCode/three/VRButton.module.js';
 import { Camera } from './Camera.js'
 import AnchoredGroup from './AnchoredGroup.js'
 
 /**
- * @typedef {number} {pixels}
+ * @typedef {number} pixels
+ * @typedef {number} DeviceUnitsPerMeter
+ * @typedef {number} MetersPerSceneUnit
+ * @typedef {number} DeviceUnitsPerSceneUnit
+ * @typedef {number} SceneUnitsPerDeviceUnit
+ * @typedef {number} SceneUnits
+ * @typedef {(globalScale: GlobalScale) => void} GlobalScaleListener
  * @typedef {{a: number, b: number, c: number, normal: THREE.Vector3, materialIndex: number}} Face
- * @typedef {{distance: number, distanceToRay?: number|undefined, point: THREE.Vector3, index?: number | undefined, face?: Face | null | undefined, faceIndex?: number | undefined, object: THREE.Object3D, uv?: THREE.Vector2 | undefined, uv1?: THREE.Vector2 | undefined, normal?: THREE.Vector3, instanceId?: number | undefined, pointOnLine?: THREE.Vector3, batchId?: number}} Intersection
+ * @typedef {{distance: number, distanceToRay?: number|undefined, point: THREE.Vector3, index?: number | undefined, face?: Face | null | undefined, faceIndex?: number | undefined, object: THREE.Object3D, uv?: THREE.Vector2 | undefined, uv1?: THREE.Vector2 | undefined, normal?: THREE.Vector3, instanceId?: number | undefined, pointOnLine?: THREE.Vector3, batchId?: number, scenePoint: THREE.Vector3, sceneDistance: SceneUnits}} Intersection
  */
 
+class GlobalScale {
+    /** @type {DeviceUnitsPerMeter} */
+    #deviceScale;
+
+    /** @type {MetersPerSceneUnit} */
+    #sceneScale;
+
+    /** @type {THREE.Group} */
+    #node;
+
+    /** @type {GlobalScaleListener[]} */
+    #listeners;
+    
+    /**
+     * 
+     * @param {DeviceUnitsPerMeter} deviceScale 
+     * @param {MetersPerSceneUnit} sceneScale 
+     */
+    constructor(deviceScale, sceneScale) {
+        this.#deviceScale = deviceScale;
+        this.#sceneScale = sceneScale;
```

Comment:
the scene scale doesn't actually change, but a lot of code depends on the separation of device scale and scene scale, this could be a constant. But should it be, maybe in the remote operator we want to be able to scale environments if we want to display multiple environments at once (minimap idea of the hackathon). Most of the code would already be compatible.